compiler/codegen/expr.cpp

```diff
@@ -4044,6 +4044,44 @@ DEFINE_PRIM(PRIM_ASSIGN) {
       codegenAssign(lhs, rg);
     }
 }
+DEFINE_PRIM(PRIM_ASSIGN_UNORDERED) {
+
+  Expr* lhsExpr = call->get(1);
+  Expr* rhsExpr = call->get(2);
+  bool lhsWide = lhsExpr->isWideRef();
+  bool rhsWide = rhsExpr->isWideRef();
+
+  if (lhsWide == false && rhsWide == true) {
+    // do an unordered GET
+    // chpl_comm_get_unordered(
+    //   void *dst,
+    //   c_nodeid_t src_node, void* src_raddr,
+    //   size_t size, int32_t typeIndex, int32_t commID,
+    //   int ln, int32_t fn);
+    GenRet dst = codegenValuePtr(call->get(1));
+    GenRet src = call->get(2);
+    GenRet ln = call->get(3);
+    GenRet fn = call->get(4);
+    TypeSymbol* dt = call->get(1)->typeInfo()->getValType()->symbol;
+    GenRet size = codegenSizeof(dt->typeInfo());
+
+    if (call->get(1)->typeInfo()->symbol->hasFlag(FLAG_REF))
+      dst = codegenDeref(dst);
+
+    codegenCall("chpl_comm_get_unordered",
```

Comment:
If possible, it'd be nice to reuse the logic in `codegenPutGet` or somehow call that function

Reply:
I'd like to leave it this way for these reasons:
 * codegenPutGet is much more complicated (it handles more cases) and at this time I'd prefer PRIM_UNORDERED_ASSIGN to be simple rather than general (since we have pretty limited compiler optimization and runtime support).
 * in the future, I hope we replace PRIM_CHPL_COMM_GET_UNORDERED/PRIM_CHPL_COMM_GET/PRIM_CHPL_COMM_PUT/codegenPutGet with something for codegenning a PRIM_MEMCPY with possibly remote data. In that event, codegenPutGet and the related primitives won't exist anyway.